Error Compiling

Hi,

First of all thanks for this project, I'm trying to tinker around and upload your code into a nodeMCU v3 (esp-8266). Although, I'm having some issue when trying to compile the code, and can't figure what's wrong.

The errors seems to be in the Debug.h file. Possibly some issue with my environment, but any help is thanked.

I'm leaving here the logs in case you want to check it out.

> Indexing .pio\build\d1_mini\lib6f2\libAHRS.a
> Archiving .pio\build\d1_mini\libb39\libEspSoftSerial.a
> Compiling .pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFi.cpp.o
> Compiling .pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiAP.cpp.o
> Archiving .pio\build\d1_mini\libaa4\libEspWire.a
> Archiving .pio\build\d1_mini\lib9c2\libEscDriver.a
> Indexing .pio\build\d1_mini\libb39\libEspSoftSerial.a
> Indexing .pio\build\d1_mini\libaa4\libEspWire.a
> Indexing .pio\build\d1_mini\lib9c2\libEscDriver.a
> Archiving .pio\build\d1_mini\libb6e\libWire.a
> Compiling .pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiGeneric.cpp.o
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from lib\Espfc\src/Device/InputPPM.h:5,
>                  from lib\Espfc\src\Device\InputPPM.cpp:1:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> Indexing .pio\build\d1_mini\libb6e\libWire.a
> Compiling .pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiGratuitous.cpp.o
> Archiving .pio\build\d1_mini\lib618\libSPI.a
> Compiling .pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiMulti.cpp.o
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\EEPROM\EEPROM.cpp:22:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/SerialManager.h: In member function 'int Espfc::SerialManager::begin()':
> lib\Espfc\src/debug.h:21:28: error: '_debugStream' was not declared in this scope
>    21 | #define LOG_SERIAL_INIT(p) _debugStream = p;
>       |                            ^~~~~~~~~~~~
> lib\Espfc\src/SerialManager.h:104:11: note: in expansion of macro 'LOG_SERIAL_INIT'
>   104 |           LOG_SERIAL_INIT(port)
>       |           ^~~~~~~~~~~~~~~
> Indexing .pio\build\d1_mini\lib618\libSPI.a
> Compiling .pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiSTA-WPS.cpp.o
> *** [.pio\build\d1_mini\libc0d\Espfc\Device\InputPPM.cpp.o] Error 1
> *** [.pio\build\d1_mini\lib602\EEPROM\EEPROM.cpp.o] Error 1
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\CertStoreBearSSL.h:23,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\CertStoreBearSSL.cpp:20:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Updater.h:4,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\BearSSLHelpers.h:28,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\BearSSLHelpers.cpp:23:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> *** [.pio\build\d1_mini\src\main.cpp.o] Error 1
> *** [.pio\build\d1_mini\lib156\ESP8266WiFi\CertStoreBearSSL.cpp.o] Error 1
> *** [.pio\build\d1_mini\lib156\ESP8266WiFi\BearSSLHelpers.cpp.o] Error 1
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\WiFiClient.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFi.h:39,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFiAP.cpp:25:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\WiFiClient.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFi.h:39,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFi.cpp:25:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\WiFiClient.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFi.h:39,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFiGeneric.cpp:29:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> *** [.pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiAP.cpp.o] Error 1
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\WiFiClient.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFi.h:39,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFiMulti.h:30,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFiMulti.cpp:27:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> *** [.pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFi.cpp.o] Error 1
> In file included from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Stream.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/HardwareSerial.h:32,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\cores\esp8266/Arduino.h:288,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\WiFiClient.h:25,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFi.h:39,
>                  from C:\Users\{user}\.platformio\packages\framework-arduinoespressif8266\libraries\ESP8266WiFi\src\ESP8266WiFiSTA-WPS.cpp:26:
> lib\Espfc\src/debug.h:19:8: error: 'Stream' does not name a type; did you mean 'Stream_h'?
>    19 | static Stream * _debugStream = nullptr;
>       |        ^~~~~~
>       |        Stream_h
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T)':
> lib\Espfc\src/debug.h:27:6: error: '_debugStream' was not declared in this scope
>    27 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> lib\Espfc\src/debug.h: In function 'void Espfc::D(T, Args ...)':
> lib\Espfc\src/debug.h:37:6: error: '_debugStream' was not declared in this scope
>    37 |   if(_debugStream)
>       |      ^~~~~~~~~~~~
> *** [.pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiMulti.cpp.o] Error 1
> *** [.pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiSTA-WPS.cpp.o] Error 1
> *** [.pio\build\d1_mini\lib156\ESP8266WiFi\ESP8266WiFiGeneric.cpp.o] Error 1

Thanks

--- a/lib/Espfc/src/Debug_Espfc.h
+++ b/lib/Espfc/src/Debug_Espfc.h
@@ -1,8 +1,8 @@
 #ifndef _ESPFC_DEBUG_H_
 #define _ESPFC_DEBUG_H_
 
-#include <EspGpio.h>
 #include <Arduino.h>
+#include <EspGpio.h>
 
 namespace Espfc
 {

--- a/lib/Espfc/src/Device/BaroBMP085.h
+++ b/lib/Espfc/src/Device/BaroBMP085.h
@@ -3,7 +3,7 @@
 
 #include "BusDevice.h"
 #include "BaroDevice.h"
-#include "Debug.h"
+#include "Debug_Espfc.h"
 
 #define BMP085_DEFAULT_ADDRESS 0x77
 

--- a/lib/Espfc/src/Device/BaroBMP280.h
+++ b/lib/Espfc/src/Device/BaroBMP280.h
@@ -3,7 +3,7 @@
 
 #include "BusDevice.h"
 #include "BaroDevice.h"
-#include "Debug.h"
+#include "Debug_Espfc.h"
 
 #define BMP280_DEFAULT_ADDRESS        0x77
 #define BMP280_WHOAMI_ID              0x58

--- a/lib/Espfc/src/Device/BusSPI.h
+++ b/lib/Espfc/src/Device/BusSPI.h
@@ -2,7 +2,7 @@
 #define _ESPFC_DEVICE_BUSSPI_H_
 
 #include "BusDevice.h"
-#include "Debug.h"
+#include "Debug_Espfc.h"
 #include <SPI.h>
 
 namespace Espfc {

--- a/lib/Espfc/src/Device/GyroMPU6050.h
+++ b/lib/Espfc/src/Device/GyroMPU6050.h
@@ -4,7 +4,7 @@
 #include "BusDevice.h"
 #include "GyroDevice.h"
 #include "helper_3dmath.h"
-#include "Debug.h"
+#include "Debug_Espfc.h"
 
 #define MPU6050_ADDRESS_AD0_LOW     0x68 // address pin low (GND), default for InvenSense evaluation board
 #define MPU6050_ADDRESS_AD0_HIGH    0x69 // address pin high (VCC)

--- a/lib/Espfc/src/Device/GyroMPU9250.h
+++ b/lib/Espfc/src/Device/GyroMPU9250.h
@@ -4,7 +4,7 @@
 #include "BusDevice.h"
 #include "GyroMPU6050.h"
 #include "helper_3dmath.h"
-#include "Debug.h"
+#include "Debug_Espfc.h"
 
 #define MPU9250_USER_CTRL         0x6A
 #define MPU9250_I2C_MST_EN        0x20

--- a/lib/Espfc/src/Logger.h
+++ b/lib/Espfc/src/Logger.h
@@ -2,7 +2,7 @@
 #define _ESPFC_LOGGER_H_
 
 #include <Arduino.h>
-#include "Debug.h"
+#include "Debug_Espfc.h"
 
 #if defined(ESP32)
 #include "SPIFFS.h"

--- a/lib/Espfc/src/Model.h
+++ b/lib/Espfc/src/Model.h
@@ -8,7 +8,7 @@
 #include <EscDriver.h>
 #include <Hal.h>
 
-#include "Debug.h"
+#include "Debug_Espfc.h"
 #include "ModelConfig.h"
 #include "ModelState.h"
 #include "Storage.h"

